odoo/fields.py

```diff
@@ -3007,8 +3012,30 @@ def __get__(self, records, owner=None):
 
     def setup_nonrelated(self, model):
         super().setup_nonrelated(model)
```

Comment:
For related fields, we should also check typing information for consistency.
- `my_field = fields.Many2one[base.ResPartner](related='create_uid')` - this should raise an error.
- `my_field = fields.Many2one[base.ResPartner](comodel='res.users')` - this should raise an error.